// vite-react/dist/index.js
"use strict";Object.defineProperty(exports, "__esModule", {value: true}); function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { newObj[key] = obj[key]; } } } newObj.default = obj; return newObj; } } function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }// src/index.ts
var _vite = require('vite'); var vite = _interopRequireWildcard(_vite);


// src/vite.ts


// src/handler/build.ts
var _path = require('path');
var assetPatterns = [
  ["media", /\.(mp4|webm|ogg|mp3|wav|flac|aac|swf)(\?.*)?$/i],
  ["image", /\.(png|jpe?g|gif|ico|svg|webp)(\?.*)?$/i],
  ["font", /\.(woff2?|eot|ttf|otf)(\?.*)?$/i],
  ["style", /\.(s?css|less|styl)(\?.*)?$/i]
];
var handleBuild = (config, _env) => {
  config.build ||= {};
  config.build.reportCompressedSize ??= false;
  config.build.rollupOptions ||= {};
  config.build.rollupOptions.output ||= {};
  const keys = Object.keys(config.build.rollupOptions.output);
  const overrideKeys = [
    "assetFileNames",
    "chunkFileNames",
    "entryFileNames"
  ];
  const userDefined = keys.some((key) => overrideKeys.includes(key));
  if (!userDefined) {
    config.build.rollupOptions.output = {
      ...config.build.rollupOptions.output,
      assetFileNames(assetInfo) {
        const ext = _path.extname.call(void 0, assetInfo.name || "");
        let folder = "misc";
        for (const [name, pattern] of assetPatterns) {
          if (pattern.test(ext)) {
            folder = name;
            break;
          }
        }
        return `bundle/${folder}/[name]-[hash][extname]`;
      },
      chunkFileNames: "bundle/js/[name]-[hash].js",
      entryFileNames: "bundle/js/[name]-[hash].js"
    };
  }
};

// src/handler/css.ts
var handleCss = (config, env) => {
  config.css ||= {};
  config.css.modules ||= {};
  if (config.css.modules !== false) {
    config.css.modules.generateScopedName ??= env.command === "build" ? "[hash:base64]" : "[path][name]__[local]";
  }
};

// src/handler/html.ts
var _vitepluginhtml = require('vite-plugin-html');
var handleHtml = (config, env) => {
  const html = config.html;
  if (!html) {
    return;
  }
  let minify = html.minify;
  if (minify === void 0) {
    minify = env.command === "build";
  }
  config.plugins ||= [];
  config.plugins.push(_vitepluginhtml.createHtmlPlugin.call(void 0, {
    minify,
    inject: {
      data: html.injectData,
      tags: html.injectTags
    },
    template: html.template
  }));
};

// src/handler/legacy.ts
var _pluginlegacy = require('@vitejs/plugin-legacy'); var _pluginlegacy2 = _interopRequireDefault(_pluginlegacy);

// src/util/override.ts
var override = (fn, env, options) => {
  if (typeof fn === "object") {
    return fn;
  }
  if (typeof fn === "function") {
    const result = fn(options, env);
    return result === void 0 ? options : result;
  }
  return options;
};

// src/util/enable.ts
var enable = (fn, env, defaultEnalbe) => {
  if (typeof fn === "function") {
    return fn(env);
  }
  if (typeof fn === "boolean") {
    return fn;
  }
  if (typeof defaultEnalbe === "function") {
    return defaultEnalbe(env);
  }
  return defaultEnalbe;
};

// src/handler/legacy.ts
var handleLegacy = (config, env) => {
  var _a, _b;
  config.plugins ||= [];
  if (enable((_a = config.legacy) == null ? void 0 : _a.enable, env, false)) {
    config.plugins.push(_pluginlegacy2.default.call(void 0, override((_b = config.legacy) == null ? void 0 : _b.options, env, {})));
  }
};

// src/handler/mix.ts
var handleMix = (config, _env) => {
  config.base ??= "./";
  config.clearScreen ??= false;
};

// src/handler/react.ts
var _pluginreact = require('@vitejs/plugin-react'); var _pluginreact2 = _interopRequireDefault(_pluginreact);
var handleReact = (config, env) => {
  config.plugins ||= [];
  config.plugins.push(_pluginreact2.default.call(void 0, override(config.react, env, {})));
};

// src/handler/server.ts
var handleServer = (config, _env) => {
  config.server ||= {};
  config.server.open ??= true;
};

// src/handler/styleImport.ts


var _vitepluginstyleimport = require('vite-plugin-style-import');

var styleResolves = {
  antd: _vitepluginstyleimport.AntdResolve
};
var handleStyleImport = (config, env) => {
  var _a, _b;
  config.plugins ||= [];
  if (enable((_a = config.styleImport) == null ? void 0 : _a.enable, env, true)) {
    config.plugins.push(_vitepluginstyleimport.createStyleImportPlugin.call(void 0, override((_b = config.styleImport) == null ? void 0 : _b.options, env, {})));
  }
};

// src/vite.ts
var defineConfig = (config = {}) => {
  return _vite.defineConfig.call(void 0, (env) => {
    return parseConfig(typeof config === "function" ? config(env) : config, env);
  });
};
var parseConfig = (config, env) => {
  handleReact(config, env);
  handleLegacy(config, env);
  handleCss(config, env);
  handleStyleImport(config, env);
  handleBuild(config, env);
  handleMix(config, env);
  handleServer(config, env);
  handleHtml(config, env);
  return config;
};







exports.defineConfig = defineConfig; exports.loadConfigFromFile = _vite.loadConfigFromFile; exports.loadEnv = _vite.loadEnv; exports.mergeConfig = _vite.mergeConfig; exports.styleResolves = styleResolves; exports.vite = vite;
